### PR TITLE
Validation tests

### PR DESCRIPTION
**Pull request by Itswali**
Add the following validations:
        For the User model:
            Name must not be blank.
            PostsCounter must be an integer greater than or equal to zero.
        For the Post model:
            Title must not be blank.
            Title must not exceed 250 characters.
            CommentsCounter must be an integer greater than or equal to zero.
            LikesCounter must be an integer greater than or equal to zero.
    Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  it 'likes_counter should be greater than or equal to 0' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'Return five most recent comments for post' do
+    expect(subject.recent_comments).to eq(subject.comments.order(created_at: :desc).limit(5))
+  end
```
Comment (tresorsawasawa) [inline spec/post_spec.rb]: - [ ] As required for the project to have a corresponding unit test to any custom methods in the model, kindly ensure that you also add test(s) the `increment_user_posts_counter` method

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Itswali 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/post_spec.rb (hunk 1 above)
Comment (Itswali) [reply spec/post_spec.rb]: Included the test for increment post method

Comment (peterrobert) [reply spec/post_spec.rb]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @Itswali, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**